src/widgets/dnshole/DnsHoleControls.tsx

```diff
@@ -124,53 +134,95 @@ function DnsHoleControlsWidgetTile({ widget }: DnsHoleControlsWidgetProps) {
     return dnsList;
   };
 
-  const toggleDns = async (action: 'enable' | 'disable', appsToChange?: string[]) => {
+  const toggleDns = async (
+    action: 'enable' | 'disable',
+    appsToChange?: string[],
+    hours: number = 0,
+    minutes: number = 0
+  ) => {
+    const duration = hours * 3600 + minutes * 60;
     await mutateAsync(
       {
         action,
+        duration,
         configName,
         appsToChange,
       },
       {
         onSettled: () => {
           trpcUtils.dnsHole.summary.invalidate();
+
+          setTimeout(
+            () => {
+              trpcUtils.dnsHole.summary.invalidate();
+            },
+            (duration + 3) * 1000
+          );
         },
       }
     );
+    setAppId('');
   };
 
   return (
-    <Stack justify="space-between" h={'100%'} spacing="0.25rem">
```

Comment:
Needs h=100% for the placement of the list.
Also, I know it was written as such before, but it should be w="100%", the brackets are superfluous.

Reply:
Still needs "h=100%"